verifier-sdk/src/light_transaction.rs

```diff
@@ -190,19 +193,80 @@ impl<
 
     #[heap_neutral]
     #[inline(never)]
-    pub fn emit_indexer_transaction_event(&self) -> Result<()> {
+    pub fn emit_indexer_events(&self) -> Result<()> {
```

Comment:
Interesting that you added it to the verifier sdk.
In my mind it would be easier to add it to the Merkle tree as part of the insert_leaves instruction.

Moving the changelog event logic to the Merkle tree would also be more in line with the notion to build an account compression v2 program. Such program will need to work standalone without our verifier sdk thus need to emit events itself.

Do you see any negative tradeoffs with moving this logic into the Merkle tree and insert leaves instruction?

Reply:
There only reason I added it to the verifier SDK is consinstency and keeping all noop program CPIs in one place. I feel like keeping different noop calls in completely different places might be confusing for us. 

I don't see any tradeoffs of keeping it anywhere - Solana transactions are atomic and both `light_transaction` and `insert_two_leaves` are always called. As long as we do it in place where the Merkle tree is already loaded from bytes, we should be fine.

By "be more in line with the notion to build an account compression v2 program" you mean a use-case where people would call just the Merkle tree program without interacting with verifier?

Reply:
Yes, we will want deploy the Merkle trees into a separate program to offer them as a utility to the ecosystem even without system verifier interaction.